scripts/install/auto-install.sh

```diff
@@ -0,0 +1,147 @@
+[[ -z "${POCS_BRANCH}" ]] && export POCS_BRANCH="develop"
+
+################################################################################
+# Functions COPIED from install-helper-functions.sh
+# Print the disk location of the first arg.
+function safe_which() {
+  type -p "${1}" || /bin/true
+}
+
+# Print a separator bar of # characters.
+function echo_bar() {
+  local terminal_width
+  if [ -n "${TERM}" ] && [ -t 0 ] ; then
+    if [[ -n "$(which resize)" ]] ; then
+      terminal_width="$(resize 2>/dev/null | grep COLUMNS= | cut -d= -f2)"
+    elif [[ -n "$(which stty)" ]] ; then
+      terminal_width="$(stty size 2>/dev/null | cut '-d ' -f2)"
+    fi
+  fi
+  printf "%${terminal_width:-80}s\n" | tr ' ' '#'
+}
+################################################################################
+
+function do_sudo() {
+  if [ "$(id -u -n)" == "root" ] ; then
+    echo "Running ${*}"
+    "$@"
+  else
+    echo "Running sudo ${*}; you may be prompted for your password."
+    (set -x ; sudo "$@")
+  fi
+}
+
+function clone_or_update() {
+  local -r REPO_DIR="${1}"
+  local -r REPO_URL="${2}"
+  local -r BRANCH="${3}"
+
+  echo_bar
+
+  if [ ! -d "${REPO_DIR}/.git" ]
+  then
+    if [ -d "${REPO_DIR}" ]
+    then
+      echo 2> "
+The directory (${REPO_DIR}) already exists, but doesn't appear
+to be a valid git repository. Please remove it (or move it out
+of the way) and re-run this script.
+  "
+      exit 1
+    fi
+    echo "
+Cloning ${REPO_URL} into ${REPO_DIR}
+"
+    (set -x ; git clone "${REPO_URL}" "${REPO_DIR}")
+    cd "${REPO_DIR}"
+    git checkout "${BRANCH}"
+  else
+    echo "
+Pulling the latest software into the worktree at ${REPO_DIR}
+"
+    cd "${REPO_DIR}"
+    git fetch --all
+    git checkout "${BRANCH}"
+    git pull
+  fi
+}
+
+# Exit immediately if a command fails:
+set -e
+
+# COPIED from default-env-vars.sh
+[[ -z "${PANUSER}" ]] && export PANUSER="panoptes"      # Default user
+[[ -z "${PANDIR}" ]] && export PANDIR="/var/panoptes"   # Main Dir
+[[ -z "${PANLOG}" ]] && export PANLOG="${PANDIR}/logs"  # Log files
+[[ -z "${POCS}" ]] && export POCS="${PANDIR}/POCS"      # Main Observatory Control
+[[ -z "${PAWS}" ]] && export PAWS="${PANDIR}/PAWS"      # Web Interface
+
+if [[ "$(whoami)" != "${PANUSER}" ]] ; then
+  echo >2 "Please run this script as ${PANUSER}, not as $(whoami)"
+  exit 1
+fi
+
+# Let's assume we'll need to run apt-get install, so first run apt-get update
+# which will refresh caches used during apt-get install.
+echo_bar
+do_sudo apt-get update
+
+if [ ! -x "$(safe_which git)" ]
+then
+  echo_bar
+  echo "
+git is not installed, so installing it...
+"
+  do_sudo apt-get install -y git
+fi
+
+echo_bar
+echo "
+Ensuring that ${PANDIR} exists
+"
+if [ ! -d "${PANDIR}" ]
+then
+  do_sudo mkdir -p "${PANDIR}"
+fi
+
+echo_bar
+echo "
+Ensuring that ${PANDIR} is owned by user ${PANUSER}
+"
+do_sudo chown "${PANUSER}" "${PANDIR}"
```

Comment:
```suggestion
do_sudo chown -R "${PANUSER}:${PANUSER}" "${PANDIR}"
```

I added the recursive option because I was thinking the subdirectories existed at this point but it looks like not but the option shouldn't hurt. Also not sure if wanted the group changed or not, I usually default to changing both.

Reply:
I tried that a while ago, even had it in this code earlier today, but then I remembered that it was really very slow if the full tree was already there. I can reconsider/retest if you like.

Reply:
Interesting. It seems like there shouldn't be _that_ many files in the tree. But okay not doing it. That means are assuming if it already exists then the user has already set their permissions accordingly, which I think is fine.

Reply:
I added it, and it work. I now recall that the slow case was inside of docker.